### PR TITLE
Add million char after member XP count in levels.ts

### PR DESCRIPTION
**Pull request by marzeq**
**Please describe the changes this PR makes and why it should be merged**
As Rodry is about to hit 1 million XP, I made a quick patch that adds a "m" char if the member XP is bigger than 1 million.

**In this PR:**
- Code changes were not tested
- Strings were added/removed (new/deleted string keys)

### CODE REVIEW

File: strings/en/levels.json [hunk 1]
```diff
@@ -4,5 +4,6 @@
 	"level": "#%%rank%% | Level %%level%% | XP: %%xp%%",
 	"youIndicator": "YOU",
 	"thousand": "k",
+	"million": "m",
```
Comment (ImRodry) [inline strings/en/levels.json:7]: That's not how we do millions
```suggestion
	"million": "M",
```

**Comment by marzeq** [comment]
Also I changed it to an `if, else if, else` because doing that with the ternary operator is stupid when we have 3 cases.

File: src/commands/Info/levels.ts [hunk 2]
```diff
@@ -120,11 +120,15 @@ function fetchPage(page: number, pages: DbUser[][], getString: GetStringFunction
 		// const user = interaction.client.users.cache.get(pages[page][i].id)! //Get the user if we ever decide to change that
 		if (pages[page][i].levels) {
 			const totalXp = pages[page][i].levels!.totalXp
+			let formattedXp: string
+			if (totalXp >= 1000000) formattedXp = `${(totalXp / 1000000).toFixed(2)}${getString("million")}`
+			else if (totalXp >= 1000) formattedXp = `${(totalXp / 1000).toFixed(2)}${getString("thousand")}`
```
Comment (ImRodry) [inline src/commands/Info/levels.ts:125]: Readability pls
```suggestion
			if (totalXp >= 1_000_000) formattedXp = `${(totalXp / 1_000_000).toFixed(2)}${getString("million")}`
			else if (totalXp >= 1_000) formattedXp = `${(totalXp / 1_000).toFixed(2)}${getString("thousand")}`
```

File: src/commands/Info/levels.ts [hunk 3]
```diff
@@ -120,11 +120,15 @@ function fetchPage(page: number, pages: DbUser[][], getString: GetStringFunction
...
+			else formattedXp = totalXp.toString()
```
Comment (ImRodry) [inline src/commands/Info/levels.ts:126]:
```suggestion
			else formattedXp = `${totalXp}`
```

File: strings/en/levels.json [hunk 4]
```diff
@@ -4,5 +4,6 @@
...
 	"unranked": "#%%rank%% Unranked"
-}
+}
```
Comment (ImRodry) [inline strings/en/levels.json:9]: pls revert this

**Review by ImRodry — CHANGES_REQUESTED** [review]

**Review by ImRodry — APPROVED** [review]
Thanks for the personal patch

**Review by ImRodry — CHANGES_REQUESTED** [review]
Hold on you forgot rank.ts

**Comment by marzeq** [comment]
Oopsie fuckie wuckie

File: src/commands/Info/rank.ts [hunk 5]
```diff
@@ -40,12 +40,28 @@ const command: Command = {
...
+		else if (currentXp >= 1000) currentXpFormatted = `${(currentXp / 1000).toFixed(2)}${getString("thousand")}`
+		else currentXpFormatted = `${currentXp}`
+
+		if (totalXp >= 1_000_000) xpNeededFormatted = `${(totalXp / 1_000_000).toFixed(2)}${getString("million")}`
+		else if (totalXp >= 1000) xpNeededFormatted = `${(totalXp / 1000).toFixed(2)}${getString("thousand")}`
+		else xpNeededFormatted = `${totalXp}`
+
+		if (messageCount >= 1_000_000) messagesFormatted = `${(messageCount / 1_000_000).toFixed(2)}${getString("million")}`
+		else if (messageCount >= 1000) messagesFormatted = `${(messageCount / 1000).toFixed(2)}${getString("thousand")}`
+		else messagesFormatted = `${messageCount}`
```
Comment (ImRodry) [inline src/commands/Info/rank.ts:57]: Might as well just makes this into a function tbf

Comment (marzeq) [reply src/commands/Info/rank.ts:57]: Good idea, that way it can also be shared in levels.ts

Comment (ImRodry) [reply src/commands/Info/rank.ts:57]: Only issue with it being in util is having to pass getString, so you can probably make the function in one of the files and export it to the other one

Comment (marzeq) [reply src/commands/Info/rank.ts]: Latest commit implements this functionality

File: strings/en/levels.json [hunk 6]
```diff
@@ -3,6 +3,5 @@
 	"pageTitle": "Here is the XP leaderboard!",
 	"level": "#%%rank%% | Level %%level%% | XP: %%xp%%",
 	"youIndicator": "YOU",
-	"thousand": "k",
```
Comment (ImRodry) [inline strings/en/levels.json:6]: Please add this one (and only this one) back in order to keep the translations on crowdin

Comment (marzeq) [reply strings/en/levels.json]: Resolved in latest commit

**Review by ImRodry — APPROVED** [review]

**Comment by ImRodry** [comment]
GitHub isn't complaining but this needs a rebase

**Pull request merged**